ide/parsing.indexing/src/org/netbeans/modules/parsing/spi/indexing/ErrorsCache.java

```diff
@@ -82,6 +94,62 @@ public static interface Convertor<T> {
         public String    getMessage(T t);
     }
 
+    /**Position in a text expressed as 1-based line and character offset.
+     * @since 9.39
+     */
+     public static final class Position {
+        private int line;
+        private int column;
+
+        public Position(int line, int column) {
+            this.line = line;
+            this.column = column;
+        }
+
+        public int getLine() {
+            return line;
+        }
+
+        public int getColumn() {
+            return column;
+        }
+    }
+
+    /**Range in a text expressed as (1-based) start and end positions.
+     * @since 9.39
+     */
+    public static final class Range {
+        private Position start;
+        private Position end;
+
+        public Range(Position start, Position end) {
+            this.start = start;
+            this.end = end;
+        }
+
+        public Position getStart() {
+            return start;
+        }
+
+        public Position getEnd() {
+            return end;
+        }
+    }
+
+    /**Getter for properties of the given error description including the range.
+     * @since 9.39
+     */
+    public static interface Convertor2<T> extends Convertor<T> {
+        public Range getRange(T t);
+    }
```

Comment:
is the `Convertor2` vs `Convertor` type difference important? If not this could be a default method in `Convertor` returning null?

there is one `instanceof Convertor2` in `TaskCache` which could potentially be a null check.

Reply:
Fixed. Thanks for pointing this out.